src/gt4py/backend/dawn_backends.py

```diff
@@ -58,8 +76,23 @@ def visit_FieldDecl(self, node: gt_ir.FieldDecl, **kwargs):
             )
         )
 
+    def visit_FieldRef(self, node: gt_ir.FieldRef, **kwargs):
+        field_name = node.name
+        if self.in_write:
```

Comment:
The generic node visitors always forward the received `**kwargs` and I think is good practice for custom `visit_` method handlers to do it too. I find convenient to pass an optional keyword argument with this information to the inner `visit_` method from the outer one, for this kind of local (in/out) state tracking. For example in this case you can avoid the `self.in_write` var by just doing:

```
def visit_FieldRef(self, node: gt_ir.FieldRef, *, in_write: bool = False, **kwargs):
    ...
    if in_write:
     ....
```
and then in line 88:
```left = self.visit(node.target, in_write=True, **kwargs)```

This should work perfectly if you always pass the `**kwargs` to the dispatcher `self.visit` method: `self.visit(..., **kwargs)`

Don't take this as a request to change the code but just as a comment for your information. Feel free to keep it as it is right now if it makes more sense to you.

Reply:
Thank you Enrique, that is an excellent suggestion.